build.gradle

```diff
@@ -221,24 +227,61 @@ bundlePlugin {
     }
 }
 
-configurations.all {
-    resolutionStrategy {
-        force 'commons-codec:commons-codec:1.14'
-        force 'org.slf4j:slf4j-api:1.7.30'
-        force 'org.scala-lang:scala-library:2.13.8'
-        force 'commons-io:commons-io:2.11.0'
-        force "com.fasterxml.jackson:jackson-bom:${versions.jackson}"
-        force "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
-        force "com.fasterxml.jackson.datatype:jackson-datatype-jdk8:${versions.jackson}"
-        force "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
-        force "io.netty:netty-buffer:${versions.netty}"
-        force "io.netty:netty-common:${versions.netty}"
-        force "io.netty:netty-handler:${versions.netty}"
-        force "io.netty:netty-transport:${versions.netty}"
-        force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+configurations {
+    all {
+        resolutionStrategy {
+            force 'commons-codec:commons-codec:1.14'
+            force 'org.slf4j:slf4j-api:1.7.30'
+            force 'org.scala-lang:scala-library:2.13.8'
+            force 'commons-io:commons-io:2.11.0'
+            force "com.fasterxml.jackson:jackson-bom:${versions.jackson}"
+            force "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
+            force "com.fasterxml.jackson.datatype:jackson-datatype-jdk8:${versions.jackson}"
+            force "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
+            force "io.netty:netty-buffer:${versions.netty}"
+            force "io.netty:netty-common:${versions.netty}"
+            force "io.netty:netty-handler:${versions.netty}"
+            force "io.netty:netty-transport:${versions.netty}"
+            force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+        }
     }
+
+    integrationTestImplementation.extendsFrom implementation
+    integrationTestRuntimeOnly.extendsFrom runtimeOnly
 }
 
+//create source set 'integrationTest'
+//add classes from the main source set to the compilation and runtime classpaths of the integrationTest
+sourceSets {
+    integrationTest {
+        java {
+            compileClasspath += sourceSets.main.output
+            runtimeClasspath += sourceSets.main.output
+            srcDir file ('src/integrationTest/java')
+        }
+        resources {
+            srcDir file('src/integrationTest/resources')
+        }
+        processIntegrationTestResources {
+            duplicatesStrategy(DuplicatesStrategy.INCLUDE)
+        }
+    }
+}
+
+//add new task that runs integration tests
+task integrationTest(type: Test) {
```

Comment:
Do we need to add an additional grade command to execute these tests? I don't see them running in the github workflows

Reply:
The integrationTest task should be run only when the test task is executed, otherwise it's excluded from execution.